src/Nethermind/Nethermind.Blockchain.Test/TransactionPoolTests.cs

```diff
@@ -194,6 +197,41 @@ public void should_add_transactions_to_persistent_storage()
             transactions.Pending.Count().Should().Be(transactions.Filtered.Count());
         }
 
+        [Test]
```

Comment:
where are the tests for queuing transactions without reservations? is that still a valid use case?

how about scenarios outside of a happy path? when transaction is added, then reorganized out then comes back?